test/e2e/set/nodeconfig/nodeconfig_disksetup.go

```diff
@@ -428,11 +427,12 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
 
 				return nc
 			},
-			preNodeConfigCreationFunc: func(ctx context.Context, nc *scyllav1alpha1.NodeConfig) func(context.Context) {
+			preNodeConfigCreationFunc: func(ctx context.Context, nc *scyllav1alpha1.NodeConfig, nodeUnderTest *corev1.Node) func(context.Context) {
```

Comment:
Did you consider passing just a resource name instead of the whole node resource?

I am not familiar enough with the code to make a judgment call, but passing an entire resource to the deferred function where just a name is needed might lead to (unjustified) implied guarantee of the resource contents being up-to-date.